### PR TITLE
Adding Mellon python package

### PR DESCRIPTION
**Pull request by katosh**
This recipe adds the python package [Mellon](https://github.com/settylab/Mellon) that we developed at the [SettyLab](https://research.fredhutch.org/setty/en.html).

There are no direct non-python dependencies so this will be for @conda-forge/help-python.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by ManuSetty** [comment]
Including me as a maintainer

**Comment by katosh** [comment]
@dopplershift could the failing widows build be ignored because ax doesn't have a windows install candidate? Like, e.g., in
https://github.com/conda-forge/staged-recipes/pull/20548 and https://github.com/conda-forge/staged-recipes/pull/20551

File: recipes/mellon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - mellon
+  commands:
+    - pip check
+  requires:
+    - pip
+
+about:
+  home: https://github.com/settylab/mellon
+  summary: Non-parametric density estimator.
+  license: GPL-2.0-or-later
```
Comment (dopplershift) [inline recipes/mellon/meta.yaml:38]: Looks like it's actually:
```suggestion
  license: GPL-3.0-or-later
```

**Review by dopplershift — CHANGES_REQUESTED** [review]
Just a minor tweak to the license metadata and this is good to go. We can ignore the noarch build failure on Windows due to missing jax.

**Comment by katosh** [comment]
Thank you @dopplershift ! I included the suggestion.

**Review by dopplershift — APPROVED** [review]

**Comment by dopplershift** [comment]
Ignoring noarch Windows build failure due to missing dependency.

**Pull request merged**